src/stores/sync/sync-slice.ts

```diff
@@ -0,0 +1,131 @@
+import { createSlice } from '@reduxjs/toolkit';
```

Comment:
I haven't finished figuring out all the code around `sync` store, but I am thinking - would it be less code and woudl it be more clear and simple code if we use `react-query` here? I have feeling that redux here is redundant. 
WDYT?